model/PermissionProvider.php

```diff
@@ -136,4 +137,9 @@ public static function getSupportedRootClasses()
             new core_kernel_classes_Class(GenerisRdf::CLASS_ROLE)
         ];
     }
+
+    public function getResourceAccessData(core_kernel_classes_Resource $resource): array
+    {
+        return AdminService::getUsersPermissions($resource->getUri());
```

Comment:
Good improvement, but moving the functionality to getResourceAccessData() and referencing it from AdminService::getUsersPermissions() for backward compatibility would be cleaner (as it removes the `ServiceManager::getServiceManager()` from the chain)

Reply:
Tried to move things and added a test for the new method. Please have a look